dts/bindings/dma/gd,gd32-dma.yaml

```diff
@@ -4,6 +4,79 @@
 description: |
   GD32 DMA controller
 
+  The node has 4-cells.
+
+  channel: Select channel for data transmitting
+  slot: Select peripheral to connect DMA (Effective only for GD32F4xx series)
+  channel-config: A 32bit mask specifying the DMA channel configuration
+      - bit 6-7: Direction  (see dma.h)
+                 - 0x0: MEMORY to MEMORY
+                 - 0x1: MEMORY to PERIPH
+                 - 0x2: PERIPH to MEMORY
+                 - 0x3: reserved for PERIPH to PERIPH
+
+        - bit 9: Peripheral address increase
+                 - 0x0: no address increment between transfers
+                 - 0x1: increment address between transfers
+
+       - bit 10: Memory address increase
+                 - 0x0: no address increase between transfers
+                 - 0x1: increase address between transfers
+
+    - bit 11-12: Peripheral data width
+                 - 0x0: 8 bits
+                 - 0x1: 16 bits
+                 - 0x2: 32 bits
+                 - 0x3: reserved
+
+    - bit 13-14: Memory data width
+                 - 0x0: 8 bits
+                 - 0x1: 16 bits
+                 - 0x2: 32 bits
+                 - 0x3: reserved
+
+       - bit 15: Peripheral Increment Offset Size
+                 - 0x0: offset size is linked to the peripheral bus width
+                 - 0x1: offset size is fixed to 4 (32-bit alignment)
+
+    - bit 16-17: Priority
+                 - 0x0: low
+                 - 0x1: medium
+                 - 0x2: high
+                 - 0x3: very high
+
+  features: A 32bit bitfield value specifying DMA features
+      - bit 0-1: DMA FIFO threshold (Effective only for GD32F4xx series)
+                 - 0x0: 1/4 full FIFO
+                 - 0x1: 1/2 full FIFO
+                 - 0x2: 3/4 full FIFO
+                 - 0x3: full FIFO
+
+
+  Example of devicetree configuration
+
+  &spi0 {
+        status = "okay";
+        pinctrl-0 = <&spi0_default>;
+        pinctrl-names = "default";
+        cs-gpios = <&gpioa 4 GPIO_ACTIVE_LOW>;
+
+        dmas = <&dma1 0 3 0 0>, <&dma1 5 3 (GD32_DMA_PRIORITY_HIGH) 0>
+        dma-names = "rx", "tx";
+  };
+
+  "spi0" uses dma1 for transmitting and receiving in the example.
+  Each is named "rx" and "tx".
+  The first cell assign channel 3 to receive, channel 5 assigned to transmit.
+  The second cell is slot. Both channels select 3.
+  Slot is effective only for GD32F4XX series. ignored in otherwise.
+  What the slot number '3' meaning depends on the DMA controller and channel.
+  See Hardware manual.
```

Comment:
Note: it seems that we're in front of different DMA IPs if that is true, have you considered using different compatibles (as ST does)? Not sure if it is worth here, but if F4XX series is the only one requiring `slot`, it could have a special binding with slot cell.

Reply:
@gmarull , @cameled 

I split the configuration that is for F4xx.

- gd,gd32-dma  (channel, channel-config)
This is a basic configuration.
The stm32 implementation, this configuration is named gd,gd32-dma-v2bis.
It is a confusable name.
I'd like to keep it simple.
But it might be better to match the stm32 naming.
I can change the name quickly if so on.

I would like to hear your opinion on this point. > @gmarull @cameled @nandojve 


- gd,gd32-dma-v1 [channel, slot, channel-config, feature]
This is configuration for F4XX series.
This is following stm,stm32-dma-v1 cells specs.

For reference, stm32-dma has three configurations.

- stm,stm32-dma-v1 [channel, slot, channel-config, feature]
- stm,stm32-dma-v2 [channel, slot, channel-config]
- stm,stm32-dma-v2bis [channel, channel-config]

(slot is in an awkward position...)